backend/antibiootit/src/test/java/fi/tuni/koodimankelit/antibiootit/controller/DiagnosesTest.java

```diff
@@ -1,12 +1,68 @@
 package fi.tuni.koodimankelit.antibiootit.controller;
 
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
 import org.junit.jupiter.api.Test;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MvcResult;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import fi.tuni.koodimankelit.antibiootit.database.data.DiagnoseInfo;
+import fi.tuni.koodimankelit.antibiootit.models.Diagnoses;
 
+/**
+ * Tests /diagnoses endpoint HTTP requests and responses
+ */
 public class DiagnosesTest extends AntibioticsControllerTest {
     
     @Test
     public void diagnosesShouldReturnList() throws Exception {
 
-        // TODO Implement test using mockMvc
+        DiagnoseInfo diagnose1 = new DiagnoseInfo(
+            "Code 1", "Diagnose 1", "Etiology 1", new ArrayList<>());
+        DiagnoseInfo diagnose2 = new DiagnoseInfo(
+            "Code 2", "Diagnose 2", "Etiology 2", new ArrayList<>());
+        List<DiagnoseInfo> diagnoseInfos = Arrays.asList(diagnose1, diagnose2);
+        Diagnoses diagnoses = new Diagnoses(diagnoseInfos);
+        
+        // Mock response
+        when(service.getAllDiagnoseInfos()).thenReturn(diagnoses);
+
+        MvcResult result = mockMvc.perform(
+            // Request
+            get("/api/antibiotics/diagnoses")
```

Comment:
HTTP API Key authorization headers should be included here, as used in `/dose-calculation`. The base class provides this functionality by  protected `getHeaders(): HttpHeaders` method.

Here's a snippet how to use it:

```suggestion
            get("/api/antibiotics/diagnoses")
            .headers(getHeaders())
```

The tests are working as they are, and I had to investigate this further. After all, this is kind of a false positive. Here's ChatGPT's explanation of why the test returns `200 OK` instead of `401 Unauthorized` when the authorization headers are missing:


![image](https://user-images.githubusercontent.com/13484230/224108547-4489a277-393a-490d-bb53-bcd57966e99f.png)